presto-main/src/main/java/com/facebook/presto/metadata/FunctionManager.java

```diff
@@ -71,11 +75,15 @@ public void addFunctions(List<? extends SqlFunction> functions)
      */
     public FunctionHandle resolveFunction(Session session, QualifiedName name, List<TypeSignatureProvider> parameterTypes)
     {
-        // TODO Actually use session
         // Session will be used to provide information about the order of function namespaces to through resolving the function.
         // This is likely to be in terms of SQL path. Currently we still don't have support multiple function namespaces, nor
         // SQL path. As a result, session is not used here. We still add this to distinguish the two versions of resolveFunction
         // while the refactoring is on-going.
+
+        if (!session.getSystemProperty("tdigest_enabled", Boolean.class) && isTDigestFunction(name.toString(), parameterTypes)) {
+            throw new PrestoException(INVALID_SESSION_PROPERTY, "Session property for tdigest functions is disabled");
```

Comment:
This error code is inaccurate. It's not an invalid session property, it's that the functionality is disabled. You can just return `FUNCTION_NOT_FOUND`.